### PR TITLE
Adds the fluentd output plugin for Datadog

### LINKED ISSUE #797: Adding Datadog output fluentd plugin
### Is your feature request related to a problem? Please describe.

I know that we can use `customPlugins` on the FluentdSpec, but this is only helpful for plugins that are packaged with the kubesphere/fluentd image.  In my case, I require the output plugin for [Datadog](https://github.com/DataDog/fluent-plugin-datadog).  

How can I install this plugin? I could use a third party image, but I think this will break the auto-reload config functionality that is baked into the kubesphere image.

Also - where is the source for the kubesphere/fluentd images?

### Describe the solution you'd like

A way to install third party fluentd plugins.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by joshuabaird**
### What this PR does / why we need it:

This PR adds support for the Fluentd [datadog](https://github.com/DataDog/fluent-plugin-datadog) output plugin.

### Which issue(s) this PR fixes:
Fixes #797 

### Does this PR introduced a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@joshuabaird Thanks for this PR.
Would you help to take a look at the CI error? https://github.com/fluent/fluent-operator/actions/runs/5316042135/jobs/9628231398?pr=803#step:7:65

Also you'll need to sign your commit:
```shell
git commit -s --amend
git push -f
```

**Comment by wenchajun** [comment]
> > https://github.com/fluent/fluent-operator/actions/runs/5316042135/jobs/9628231398?pr=803#step:7:65
> 
> Yep - will try to look at this today. Not quite sure why the test is failing yet.
> 
> Quick question for you -- how should we handle buffering options for each output plugin? Like other output_buffer plugins, this plugin supports a buffering config -- but I don't see that this has been implemented in other output plugins:
> 
> ```
>   <buffer>
>           @type memory
>           flush_thread_count 4
>           flush_interval 3s
>           chunk_limit_size 5m
>           chunk_limit_records 500
>   </buffer>
> ```
You can refer to https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/expected/fluentd-cluster-cfg-output-buffer-example.cfg#LL28C8-L28C8

File: apis/fluentd/v1alpha1/tests/tools.go [hunk 1]
```diff
@@ -399,6 +399,20 @@ spec:
...
+kind: ClusterOutput
+metadata:
+  name: fluentd-output-datadog
+  labels:
+    output.fluentd.fluent.io/enabled: "true"
+spec: 
+  outputs: 
+  - datadog:
+      dd_source: kubernetes
+	  dd_sourcecategory: kubernetes
```
Comment (wenchajun) [inline apis/fluentd/v1alpha1/tests/tools.go:414]:
```
ddSourceCategory: kubernetes
ddSource: kubernetes
```

**Comment by joshuabaird** [comment]
> > > https://github.com/fluent/fluent-operator/actions/runs/5316042135/jobs/9628231398?pr=803#step:7:65
> > 
> > 
> > Yep - will try to look at this today. Not quite sure why the test is failing yet.
> > Quick question for you -- how should we handle buffering options for each output plugin? Like other output_buffer plugins, this plugin supports a buffering config -- but I don't see that this has been implemented in other output plugins:
> > ```
> >   <buffer>
> >           @type memory
> >           flush_thread_count 4
> >           flush_interval 3s
> >           chunk_limit_size 5m
> >           chunk_limit_records 500
> >   </buffer>
> > ```
> 
> You can refer to https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/expected/fluentd-cluster-cfg-output-buffer-example.cfg#LL28C8-L28C8

Ok - so basically, we can use the `BufferCommon` plugin to specify the buffer options for the Datadog output?

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
> Ok - so basically, we can use the `BufferCommon` plugin to specify the buffer options for the Datadog output?

I think so. @wenchajun

**Pull request merged**